src/framebuffer.rs

```diff
@@ -0,0 +1,707 @@
+        &mut self.data
+    }
+}
+
+impl<C, BO, const WIDTH: usize, const HEIGHT: usize, const N: usize>
+    Framebuffer<C, C::Raw, BO, WIDTH, HEIGHT, N>
+where
+    C: PixelColor + From<C::Raw>,
+    BO: ByteOrder,
+    for<'a> RawDataSlice<'a, C::Raw, BO>: IntoIterator<Item = C::Raw>,
+{
+    /// Returns an image based on the framebuffer content.
+    pub fn as_image(&self) -> ImageRaw<'_, C, BO> {
+        ImageRaw::new(&self.data, WIDTH as u32)
+    }
+
+    /// Get pixel color.
+    ///
+    /// Returns `None` is `p` is outside the framebuffer bounding box.
+    pub fn pixel(&self, p: Point) -> Option<C> {
+        self.as_image().pixel(p)
+    }
+}
+
+macro_rules! impl_bit {
+    ($raw_type:ident) => {
+        impl<C, BO, const WIDTH: usize, const HEIGHT: usize, const N: usize>
+            Framebuffer<C, $raw_type, BO, WIDTH, HEIGHT, N>
+        where
+            C: PixelColor + Into<$raw_type>,
+        {
+            /// Sets the color of a pixel.
+            ///
+            /// Trying to set a pixel outside the framebuffer is a noop.
+            pub fn set_pixel(&mut self, p: Point, c: C) {
+                if let (Ok(x), Ok(y)) = (usize::try_from(p.x), usize::try_from(p.y)) {
+                    if x < WIDTH && y < HEIGHT {
+                        let pixels_per_bit = 8 / C::Raw::BITS_PER_PIXEL;
+                        let bits_per_row = WIDTH * C::Raw::BITS_PER_PIXEL;
+                        let bytes_per_row = (bits_per_row + 7) / 8;
+                        let byte_index = bytes_per_row * y + (x / pixels_per_bit);
+                        let bit_index = 8 - (x % pixels_per_bit + 1) * C::Raw::BITS_PER_PIXEL;
+
+                        let mask = !($raw_type::new(0xFF).into_inner() << bit_index);
+                        let bits = c.into().into_inner() << bit_index;
+
+                        self.data[byte_index] = self.data[byte_index] & mask | bits;
+                    }
+                }
+            }
+        }
+
+        impl<C, BO, const WIDTH: usize, const HEIGHT: usize, const N: usize> DrawTarget
+            for Framebuffer<C, $raw_type, BO, WIDTH, HEIGHT, N>
+        where
+            C: PixelColor<Raw = $raw_type> + Into<$raw_type>,
+        {
+            type Color = C;
+            type Error = Infallible;
+
+            fn draw_iter<I>(&mut self, pixels: I) -> Result<(), Self::Error>
+            where
+                I: IntoIterator<Item = Pixel<Self::Color>>,
+            {
+                for Pixel(p, c) in pixels {
+                    self.set_pixel(p, c);
+                }
+
+                Ok(())
+            }
+        }
+    };
+}
+
+impl_bit!(RawU1);
+impl_bit!(RawU2);
+impl_bit!(RawU4);
+
+impl<C, BO, const WIDTH: usize, const HEIGHT: usize, const N: usize>
+    Framebuffer<C, RawU8, BO, WIDTH, HEIGHT, N>
+where
+    C: PixelColor + Into<RawU8>,
+{
+    /// Sets the color of a pixel.
+    ///
+    /// Trying to set a pixel outside the framebuffer is a noop.
```

Comment:
Slightly better?

```suggestion
    /// Setting a pixel outside the framebuffer's bounding box will be a noop.
```

Or maybe something like `framebuffer's drawing area`?

Reply:
It is better. I wasn't happy with that sentence, but couldn't come up with something better.